cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -985,16 +988,25 @@ func shouldDeleteNode(ctx *controllerContext, node *v1.Node, getInstance func(*c
 }
 
 func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
-	errs := []error{}
 	deletedPods := []string{}
 
-	podList, err := ctx.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName})
+	podList, err := ctx.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName, ResourceVersion: "0"})
```

Comment:
what's the rationale for this change?

Reply:
This is necessary to ensure that the pod list will be retrieved from the api-server cache instead of going to etcd.
Setting `resourceVersion=0` ensures the pod will be returned from cache - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L621

 We identified some scalability regressions in large clusters without this change, since many nodes could be retrieving the pod list simultaneously which resulted in high load on api-server. With this change, running scalability tests showed large improvement.